File: model/src/main/java/io/github/chriszhong/model/organization/event/HasEventFactory.java
package io.github.chriszhong.model.organization.event;

import io.github.runtimemodels.chazm.relation.Has;

/**
 * The {@linkplain HasEventFactory} interface defines the API for constructing {@linkplain HasEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface HasEventFactory {

	/**
	 * Constructs an {@linkplain HasEvent}.
	 *
	 * @param category
	 *            the {@linkplain EventCategory}.
	 * @param has
	 *            the {@linkplain Has}.
	 *
	 * @return a {@linkplain HasEvent}.
	 */
	HasEvent build(EventCategory category, Has has);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/relation/HasEventFactory.kt
package runtimemodels.chazm.model.relation

import runtimemodels.chazm.api.relation.Has
import runtimemodels.chazm.model.event.EventType

/**
 * The [HasEventFactory] interface defines the API for constructing [HasEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface HasEventFactory {
    /**
     * Constructs an [HasEvent].
     *
     * @param category the [EventType].
     * @param has      the [Has].
     * @return a [HasEvent].
     */
    fun build(category: EventType, has: Has): HasEvent
}
